src/panels/lovelace/cards/hui-markdown-card.ts

```diff
@@ -114,17 +114,16 @@ export class HuiMarkdownCard extends LitElement implements LovelaceCard {
     if (changedProps.has("_config")) {
       this._tryConnect();
     }
-    if (
-      this._templateResult &&
-      this._templateResult.result.length === 0 &&
-      this._config.show_empty === false
-    ) {
-      if (!this.hidden) {
+    if (this._templateResult && this._config.show_empty === false) {
+      if (this._templateResult.result.length === 0 && !this.hidden) {
         this.style.display = "none";
         this.toggleAttribute("hidden", true);
         fireEvent(this, "card-visibility-changed", { value: false });
+      } else {
+        this.style.display = "block";
+        this.toggleAttribute("hidden", false);
+        fireEvent(this, "card-visibility-changed", { value: true });
```

Comment:
This means we will now do this every time the result is not empty, we should only do this is the card is hidden and the response not empty.